src/routes/windows-protection/components/HomeView/WindowSensor/WindowSensor.js

```diff
@@ -0,0 +1,139 @@
+import React, { memo, useState, useCallback } from 'react';
+import Draggable from 'react-draggable';
+import PropTypes from 'prop-types';
+import useStyles from 'isomorphic-style-loader/useStyles';
+import get from 'lodash/get';
+import classNames from 'classnames';
+import toUpper from 'lodash/toUpper';
+import s from './WindowSensor.css';
+import openCloseSensorIcon from '../Images/open-close-sensor.svg';
+import breakSensorIcon from '../Images/break-sensor.svg';
+
+const sensorOptions = new Map([
+  [
+    'ok',
+    {
+      sensorIconClass: s.iconOk,
+      sensorInfoClass: 'bg-success',
+    },
+  ],
+  [
+    'alarm',
+    {
+      sensorIconClass: s.iconAlert,
+      sensorInfoClass: 'bg-danger',
+    },
+  ],
+  [
+    'warning',
+    {
+      sensorIconClass: s.iconWarning,
+      sensorInfoClass: 'bg-warning',
+    },
+  ],
+  [
+    'inactive',
+    {
+      sensorIconClass: s.iconInactive,
+      sensorInfoClass: 'bg-light',
+    },
+  ],
+]);
+
+function WindowSensor(props) {
+  useStyles(s);
+  const { sensorData, isEditingMode } = props;
+  const [isHover, setHover] = useState(false);
+
+  const options = { ...sensorOptions.get(sensorData.status) };
+  options.position = {
+    x: sensorData.xCoord || 0,
+    y: sensorData.yCoord || 0,
+  };
+  if (sensorData.type === 'open/close') {
+    options.sensorIcon = openCloseSensorIcon;
+  } else {
+    options.sensorIcon = breakSensorIcon;
+  }
+
+  const toggleHoverHandler = useCallback(() => {
+    setHover(oldHover => !oldHover);
+  }, []);
+
+  return (
+    <Draggable
+      bounds="parent"
+      disabled={!isEditingMode}
+      position={options.position}
+    >
+      <div
+        className={s.sensorBadge}
+        onMouseEnter={toggleHoverHandler}
+        onMouseLeave={toggleHoverHandler}
+      >
+        <img
+          alt="sensor icon"
+          className={classNames(s.sensorIcon, options.sensorIconClass)}
+          src={options.sensorIcon}
+        />
+        {!isEditingMode && (
+          <div
+            className={classNames(
+              'card',
+              options.sensorInfoClass,
+              s.sensorInfo,
+              {
+                [s.sensorInfoHover]: isHover,
+              },
+            )}
+          >
+            <div className="card-header">{`${sensorData.type} sensor on ${sensorData.window.name}`}</div>
```

Comment:
Please use `lodash/get` for things `sensorData.window.name`, otherwise your code can crash if your server fails to respond properly